### ISSUE TITLE
Instant Search: insufficient contrast on links in dark mode

### ISSUE THREAD

**Issue by bluefuton** [Automattic/jetpack#25137]
### Impacted plugin

Jetpack, Search

### Steps to Reproduce

1. On a site with Jetpack Instant Search enabled, choose dark mode from the settings: `/wp-admin/admin.php?page=jetpack-search-configure`.
2. Inspect any link with your browser accessibility tools.

### A clear and concise description of what you expected to happen.

Text contrast should be at least 4.5 : 1 to achieve a AA rating. 

### What actually happened

It is currently 4.35.

<img width="347" alt="Screen Shot 2022-07-20 at 15 43 28" src="https://user-images.githubusercontent.com/17325/179892004-2695888c-6b50-4be4-b4e2-0d52046c4bee.png">

https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Color_contrast

### Browser

Mozilla Firefox

### Other information

<img width="251" alt="Screen Shot 2022-07-20 at 15 12 48" src="https://user-images.githubusercontent.com/17325/179892147-2a629de7-2132-44c5-9f1c-f183002ad3dc.png">
<img width="647" alt="Screen Shot 2022-07-20 at 15 12 46" src="https://user-images.githubusercontent.com/17325/179892153-5bcc5f3b-1bbf-4c10-9762-6e543fdb3586.png">

### Reproducibility

Consistent
